# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/plasma-login-manager/preselect-last.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Preselect last -usually plasma-wayland- instead of first, usually any other X11 session.

Alternatively patch default state (/var/lib/plasmalogin/state.conf) if needed.

--- plasma-login-manager-6.6.1/src/frontend/greeter/qml/GreeterState.qml.vanilla	2026-02-28 16:28:09.732235937 +0100
+++ plasma-login-manager-6.6.1/src/frontend/greeter/qml/GreeterState.qml	2026-02-28 16:57:40.281677363 +0100
@@ -49,7 +49,7 @@
         } else if (lastLoggedInSessionIndex != -1) {
             return lastLoggedInSessionIndex;
         } else {
-            return 0;
+            return PlasmaLogin.SessionModel.count ? PlasmaLogin.SessionModel.count - 1 : 0;
         }
     }
 
